### DISCUSSION TITLE
getHeader and getQuery return empty objects

### DISCUSSION THREAD

**Discussion by arabyalhomsi** [uNetworking/uWebSockets.js#177] [Q&A] [answered]
I am running uWebSockets.js on a `node:10.16.2-jessie` docker machine. When I make any GET or POST requests, the method `getHeader` and `getQuery` on the `req` argument return empty objects. However, when I use forEach to loop through my header, it works and I get the header values. Why is this happening, anyone has any idea?

**Comment by coder456** [comment, marked as answer]
its used like this `req.getHeader('headername')` returns the value of the header, needs to be lowercase https://unetworking.github.io/uWebSockets.js/generated/interfaces/httprequest.html#getheader

**Comment by arabyalhomsi** [comment]
Thank you very much! worked yeah. I also messed it up because I put the following two lines under my `.get()` route. 
`app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));`
